pkg/cafs/cafs.go

```diff
@@ -132,54 +122,85 @@ type cafsStore struct {
 }
 
 type defaultFs struct {
-	store                       cafsStore
-	leafSize                    uint32
-	prefix                      string
-	zl                          zap.Logger //nolint:structcheck,unused
-	l                           log.Logger //nolint:structcheck,unused
+	store    cafsStore
+	leafSize uint32
+	l        *zap.Logger
+
+	// prefix determines a namespace for keys
+	prefix string
+	pather func(Key) string // pathing logic
```

Comment:
i agree that specifying a procedure is the most generalized abstraction ... if it's at all possible to parameterize the procedural code in this package, i'd prefer to do that for now rather than pre-emptively use the most abstract form.

slight deja-vu here from #266 #267 and #268

Reply:
the original cafs code introduced the `pather` concept, which has been later scattered under different forms. What matters is that we have one single approach. I chose the func.

I believe that adopting prefixes to form namespaces for keys is something we will have to deploy rather sooner than later, as adoption will rapidly make our current implementation of key searches quite impractical

Reply:
follow-up note: disagree - won't fix